apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -129,8 +204,48 @@ const aichatSlice = createSlice({
         chatMessage.status = status;
       }
     },
-    setAiCustomizations: (state, action: PayloadAction<AiCustomizations>) => {
-      state.currentAiCustomizations = action.payload;
+    setStartingAiCustomizations: (
+      state,
+      action: PayloadAction<{
+        levelAiCustomizationsWithVisibility: LevelAiCustomizations;
+        studentAiCustomizations: AiCustomizations;
+      }>
+    ) => {
+      const {levelAiCustomizationsWithVisibility, studentAiCustomizations} =
+        action.payload;
+
+      let reconciledAiCustomizations: AiCustomizations = {
+        botName: levelAiCustomizationsWithVisibility.botName.value,
+        temperature: levelAiCustomizationsWithVisibility.temperature.value,
+        systemPrompt: levelAiCustomizationsWithVisibility.systemPrompt.value,
+        retrievalContexts:
+          levelAiCustomizationsWithVisibility.retrievalContexts.value,
+        modelCardInfo: levelAiCustomizationsWithVisibility.modelCardInfo.value,
+      };
+
+      for (const customizationUntyped in reconciledAiCustomizations) {
+        const customization = customizationUntyped as keyof AiCustomizations;
+
+        if (
+          levelAiCustomizationsWithVisibility[customization].visibility ===
+            Visibility.EDITABLE &&
+          studentAiCustomizations[customization]
+        ) {
+          reconciledAiCustomizations = {
+            ...reconciledAiCustomizations,
```

Comment:
This is me battling Redux again:

https://github.com/code-dot-org/code-dot-org/pull/57318#discussion_r1532527865